Sources/Paywall/Paywall/Events/InternalEvents.swift

```diff
@@ -0,0 +1,284 @@
+      Logger.debug(
+        logLevel: .debug,
+        scope: .events,
+        message: "Logged Event",
+        info: eventParams
+      )
+    }
+
+		if let event = StandardEventName(rawValue: name),
+      event == .userAttributes {
+      CacheManager.shared.addUserAttributes(eventParams)
+		}
+
+		let eventData = EventData(
+      name: name,
+      parameters: JSON(eventParams),
+      createdAt: Date.init(timeIntervalSinceNow: 0).isoString
+    )
+
+		queue.addEvent(event: eventData.jsonData)
+		if handleTrigger {
+			Paywall.shared.handleTrigger(forEvent: eventData)
+		}
+
+		return eventData
+  }
+
+  private static func clean(input: Any?) -> Any? {
+    if input is NSArray {
+      return nil
+    } else if input is NSDictionary {
+      return nil
+    } else {
+      if let input = input {
+        let json = JSON(input)
+        if json.error == nil {
+          return input
+        } else {
+          if let date = input as? Date {
+            return date.isoString
+          } else if let url = input as? URL {
+            return url.absoluteString
+          } else {
+            return nil
+          }
+        }
+      }
+    }
+
+    return nil
+  }
+
+  private static func eventParams(
+    forProduct product: SKProduct?,
+    paywallInfo: PaywallInfo,
+    otherParams: [String: Any]? = nil
+  ) -> [String: Any] {
+    var output: [String: Any] = [
+      "paywall_id": paywallInfo.id,
+      "paywall_identifier": paywallInfo.identifier,
+      "paywall_slug": paywallInfo.slug,
+      "paywall_name": paywallInfo.name,
+      "paywall_url": paywallInfo.url?.absoluteString ?? "unknown",
+      "presented_by_event_name": paywallInfo.presentedByEventWithName as Any,
+      "presented_by_event_id": paywallInfo.presentedByEventWithId as Any,
+      "presented_by_event_timestamp": paywallInfo.presentedByEventAt as Any,
+      "presented_by": paywallInfo.presentedBy as Any,
+      "paywall_product_ids": paywallInfo.productIds.joined(separator: ","),
+      "paywall_response_load_start_time": paywallInfo.responseLoadStartTime as Any,
+      "paywall_response_load_complete_time": paywallInfo.responseLoadCompleteTime as Any,
+      "paywall_response_load_duration": paywallInfo.responseLoadDuration as Any,
+      "paywall_webview_load_start_time": paywallInfo.webViewLoadStartTime as Any,
+      "paywall_webview_load_complete_time": paywallInfo.webViewLoadCompleteTime as Any,
+      "paywall_webview_load_duration": paywallInfo.webViewLoadDuration as Any,
+      "paywall_products_load_start_time": paywallInfo.productsLoadStartTime as Any,
+      "paywall_products_load_complete_time": paywallInfo.productsLoadCompleteTime as Any,
+      "paywall_products_load_duration": paywallInfo.productsLoadDuration as Any
+    ]
+
+    for key in output.keys {
+      if key.contains("_load_"),
+        let output = output[key] {
+        print(key, output)
+      }
+    }
+    print( "========== _load_ vars")
```

Comment:
Can we move this to logger instead of printing?

Reply:
Probably can but I'm not actually sure why we do this. I just left it as-is. Do print statements get removed in the final build to a framework?

Reply:
I don't think so, we should definitely add a lint rule to disallow them.

Reply:
I double checked and you're right print statements aren't removed. Have moved all `print()` statements to Logger and added a custom action for SwiftLint to flag an error if it detects print (it doesn't natively have a rule for printing).